### DISCUSSION TITLE
Is there a way to insert templated lines into an existing file?

### DISCUSSION THREAD

**Discussion by rtaycher** [copier-org/copier#687] [Q&A]
Especially appending files or appending before or after a matching line?

I want to be able to insert new functions/variables in a couple of places in my code base from templates. So you could name a THING and have different THING class/registerThing/createThing code generated in several places in your code base?

Maybe a plugin/extension of copier that does this? Or any other package you know that does this? It's hard to Google

**Comment by yajo** [comment]
You can do that in just a normal template file. When copier applies the
template the 1st time, it will ask you to overwrite it. Just say no, or say
yes but fix the diff with git before committing.

On later updates, copier will try to smartly update the file from the
template, respecting the file evolution in your project as much as
possible.
